packages/modules/devices/e3dc/device.py

```diff
@@ -22,20 +23,24 @@
 log = logging.getLogger(__name__)
 
 
-def create_device(device_config: E3dc):
-    def create_bat_component(component_config: E3dcBatSetup):
+def create_device(device_config: E3dc) -> ConfigurableDevice:
+    def create_bat_component(component_config: E3dcBatSetup) -> E3dcBat:
         return E3dcBat(device_config.id,
                        component_config)
 
-    def create_counter_component(component_config: E3dcCounterSetup):
+    def create_counter_component(component_config: E3dcCounterSetup) -> E3dcCounter:
         return E3dcCounter(device_config.id,
                            component_config)
 
-    def create_inverter_component(component_config: E3dcInverterSetup):
+    def create_inverter_component(component_config: E3dcInverterSetup) -> E3dcInverter:
         return E3dcInverter(device_config.id,
                             component_config)
 
-    def update_components(components: Iterable[Union[E3dcBat, E3dcCounter, E3dcInverter]]):
+    def create_external_inverter_component(component_config: E3dcExternalInverterSetup) -> E3dcExternalInverter:
+        return E3dcExternalInverter(device_config.id,
+                                    component_config)
+
+    def update_components(components: Iterable[Union[E3dcBat, E3dcCounter, E3dcInverter]]) -> None:
```

Comment:
Hier muss dann auch noch der externe WR dazu:
```suggestion
    def update_components(components: Iterable[Union[E3dcBat, E3dcCounter, E3dcInverter, E3dcExternalInverterSetup]]) -> None:
```